[signing] Signing fails when secret key is missing even when yolo is set

### Task List

- [x] Steps to reproduce provided
- [] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. checkout https://github.com/jreleaser/helloworld-java-bin
2. configure a signing block like this
```
signing:
  pgp:
    active: ALWAYS
```
3. Make sure you don't have JRELEASER_GPG_SECRET_KEY defined
4. `mvnw verify`
5. `jreleaser assemble`
6. `jreleaser release --dry-run --yolo`
### Expected Behaviour

A local release should be performed with signing skipped.

### Actual Behaviour

An exception occurs because the PGP secret key is missing

### Environment Information

- **Operating System**: N/A
- **JReleaser Version:** 1.24.0
- **JDK Version:** N/A

```
[ERROR] == JReleaser ==
[ERROR] signing.pgp.secretKey must not be blank. Configure a value using the CLI yaml DSL, or define a System property jreleaser.gpg.secret.key, or define a JRELEASER_GPG_SECRET_KEY environment variable, or define a key/value pair in /Users/aalmiray/.jreleaser/config.toml with a key named JRELEASER_GPG_SECRET_KEY
[ERROR] Look for more details at /Users/aalmiray/dev/github/jreleaser/helloworld-java-bin/out/jreleaser/trace.log
JReleaser has not been properly configured.
signing.pgp.secretKey must not be blank. Configure a value using the CLI yaml DSL, or define a System property jreleaser.gpg.secret.key, or define a JRELEASER_GPG_SECRET_KEY environment variable, or define a key/value pair in /Users/aalmiray/.jreleaser/config.toml with a key named JRELEASER_GPG_SECRET_KEY
```

cc: @maxandersen

--- a/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/validation/signing/SigningValidator.java
+++ b/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/validation/signing/SigningValidator.java
@@ -46,6 +46,7 @@ import static org.jreleaser.model.api.signing.Signing.MINISIGN_PASSWORD;
 import static org.jreleaser.model.api.signing.Signing.MINISIGN_PUBLIC_KEY;
 import static org.jreleaser.model.api.signing.Signing.MINISIGN_SECRET_KEY;
 import static org.jreleaser.model.internal.validation.common.Validator.checkProperty;
+import static org.jreleaser.model.internal.validation.common.Validator.mergeErrors;
 import static org.jreleaser.model.internal.validation.common.Validator.resolveActivatable;
 import static org.jreleaser.util.CollectionUtils.listOf;
 import static org.jreleaser.util.Env.envKey;
@@ -73,9 +74,13 @@ public final class SigningValidator {
         boolean activeSet = signing.isActiveSet();
         resolveActivatable(context, signing, "signing", "");
 
-        validatePgp(context, mode, signing.getPgp(), errors);
-        validateCosign(context, mode, signing.getCosign(), errors);
-        validateMinisign(context, mode, signing.getMinisign(), errors);
+        Errors incoming = new Errors();
+        validatePgp(context, mode, signing.getPgp(), incoming);
+        mergeErrors(context, errors, incoming, signing.getPgp());
+        validateCosign(context, mode, signing.getCosign(), incoming);
+        mergeErrors(context, errors, incoming, signing.getCosign());
+        validateMinisign(context, mode, signing.getMinisign(), incoming);
+        mergeErrors(context, errors, incoming, signing.getMinisign());
 
         if (!activeSet) {
             boolean enabled = signing.getPgp().isEnabled() ||

--- a/core/jreleaser-model-impl/src/test/java/org/jreleaser/model/internal/validation/signing/SigningValidatorTest.java
+++ b/core/jreleaser-model-impl/src/test/java/org/jreleaser/model/internal/validation/signing/SigningValidatorTest.java
@@ -1,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2020-2026 The JReleaser authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jreleaser.model.internal.validation.signing;
+
+import org.jreleaser.logging.SimpleJReleaserLoggerAdapter;
+import org.jreleaser.model.Active;
+import org.jreleaser.model.api.JReleaserCommand;
+import org.jreleaser.model.api.JReleaserContext.Mode;
+import org.jreleaser.model.internal.JReleaserContext;
+import org.jreleaser.model.internal.JReleaserModel;
+import org.jreleaser.util.Errors;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @since 1.25.0
+ */
+class SigningValidatorTest {
+
+    @Test
+    void signingPgpValidationFailsWithoutYolo(@TempDir Path tempDir) {
+        // Given: PGP signing active but no secret key configured
+        JReleaserContext context = createContext(tempDir, false);
+        context.getModel().getSigning().setActive(Active.ALWAYS);
+        context.getModel().getSigning().getPgp().setActive(Active.ALWAYS);
+
+        Errors errors = new Errors();
+
+        // When: validate signing
+        SigningValidator.validateSigning(context, Mode.FULL, errors);
+
+        // Then: errors should be reported
+        assertTrue(errors.hasConfigurationErrors(),
+            "Expected configuration errors when PGP secret key is missing without yolo");
+    }
+
+    @Test
+    void signingPgpValidationSkippedWithYolo(@TempDir Path tempDir) {
+        // Given: PGP signing active but no secret key configured, yolo=true
+        JReleaserContext context = createContext(tempDir, true);
+        context.getModel().getSigning().setActive(Active.ALWAYS);
+        context.getModel().getSigning().getPgp().setActive(Active.ALWAYS);
+
+        Errors errors = new Errors();
+
+        // When: validate signing with yolo
+        SigningValidator.validateSigning(context, Mode.FULL, errors);
+
+        // Then: no errors (converted to warnings), and signing should be disabled
+        assertFalse(errors.hasConfigurationErrors(),
+            "Expected no configuration errors when yolo is set");
+        assertTrue(errors.hasWarnings(),
+            "Expected warnings when yolo is set and PGP secret key is missing");
+        assertFalse(context.getModel().getSigning().getPgp().isEnabled(),
+            "Expected PGP signing to be disabled when yolo is set and secret key is missing");
+    }
+
+    private static JReleaserContext createContext(Path basedir, boolean yolo) {
+        // Use a temp dir as settings path to avoid loading ~/.jreleaser/config.properties
+        // which may contain actual credentials
+        JReleaserContext context = new JReleaserContext(
+            new SimpleJReleaserLoggerAdapter(SimpleJReleaserLoggerAdapter.Level.WARN),
+            JReleaserContext.Configurer.CLI_YAML,
+            Mode.FULL,
+            JReleaserCommand.FULL_RELEASE,
+            new JReleaserModel(),
+            basedir,
+            basedir.resolve("settings.properties"),
+            basedir.resolve("out/jreleaser"),
+            yolo,
+            true,
+            true,
+            false,
+            false,
+            Collections.emptyList(),
+            Collections.emptyList());
+        // Initialize environment properties using the temp settings path
+        // (no config.properties exists there, so no external credentials leak in)
+        context.getModel().getEnvironment().initProps(context);
+        return context;
+    }
+}

